tuplex/core/src/ee/local/LocalBackend.cc

```diff
@@ -693,6 +694,31 @@ namespace tuplex {
             }
         }
 
+        // assign the order for all tasks
+        for(size_t i = 0; i < tasks.size(); ++i) {
+            tasks[i]->setOrder(i);
+        }
+
+        if (tstage->hasOutputLimit()) {
+            if (tstage->outputTopLimit() > 0 && tstage->outputBottomLimit() > 0) {
+                // do task striping for output limit on both ends
+                vector<IExecutorTask*> newTasks;
+                for(size_t i = 0; i < tasks.size() - i; i++) {
+                    const size_t rev_i = tasks.size() - 1 - i;
+                    newTasks.push_back(tasks[i]);
+                    if (i < rev_i) {
+                        newTasks.push_back(tasks[rev_i]);
+                    }
+                }
+                assert(tasks.size() == newTasks.size());
+                tasks.swap(newTasks);
+            } else if (tstage->outputBottomLimit() > 0) {
+                // bottom limit only, just reverse the task order
+                std::reverse(tasks.begin(), tasks.end());
+            }
+            // if top limit only, do nothing since the order is already good
```

Comment:
where is toplimit case only handled??

Reply:
It is a default case when the task was generated. I have added a comment to clarify this